plugins/schedule/base/lib/model/ScheduleEventPeer.php

```diff
@@ -248,9 +253,56 @@ public static function retrieveEventsByResourceIdsAndDateWindow($resourceIds, $s
 	 */
 	public static function retrieveBlackoutEventsByDateWindow($startDate, $endDate, $scheduleEventIdToIgnore = null)
 	{
+		$cacheResult = self::tryGetBlackoutResultFromCache($startDate, $endDate);
+		if(isset($cacheResult))
+		{
+			return $cacheResult;
+		}
+
 		$c = self::getRetrieveEventsByDateWindowCriteria($startDate, $endDate, $scheduleEventIdToIgnore);
 		$c->addAnd(ScheduleEventPeer::TYPE, scheduleEventType::BLACKOUT, Criteria::EQUAL);
-		return self::doSelect($c);
+		$result = self::doSelect($c);
+		self::addBlackoutResultToCache($startDate, $endDate, $result);
+		return $result;
+	}
+
+	protected static function tryGetBlackoutResultFromCache($startDate, $endDate)
+	{
+		$partnerId = kCurrentContext::getCurrentPartnerId();
```

Comment:
It is not possible that in the same session we shall have more than one partner, since this value dies when session ends,